{% set name = "pysmbc" %}
{% set version = "1.0.23" %}
{% set sha256 = "08db176281103024b7e8563adef733e56250051d87394c4baec4fd298d0d16f8" %}

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz   # The extension has changed in the latest versions to .tar.gz
  sha256: {{ sha256 }}
  patches:
    # See the bug report at https://github.com/hamano/pysmbc/issues/53 and the
    # proposed fix in https://github.com/hamano/pysmbc/pull/54.
    - 0001-setup.py-v1.0.22-install-issues.patch

build:
  number: 0
  script: |
    export LDFLAGS=-L/usr/lib64
    export CPPFLAGS=-I/usr/include
    {{ PYTHON }} -m pip install . --no-deps -vv
  skip: true  # [py<36 or not linux]

requirements:
  build:
    - {{ compiler('c') }}
    - {{ compiler('cxx') }}
  host:
    - python
    - pip
  run:
    - python

# Commenting out the tests for now as the "undefined symbol: lp_set_cmdline"
# error happens when running this test with libsmbclient v3.6. It works well
# with version 4.2 which is installed at NSLS-II ISS.
# test:
#   imports:
#     smbc

about:
  home: http://pypi.python.org/pypi/pysmbc
  license: GNU GPL v2.0
  license_file: COPYING
  summary: libsmbclient binding for Python
  dev_url: https://github.com/hamano/pysmbc
